packages/react-native-reanimated/Common/cpp/reanimated/Tools/FeatureFlags.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include <string>
+#include <unordered_map>
+
+namespace reanimated {
+
+class StaticFeatureFlags {
+ public:
+#ifdef REANIMATED_FEATURE_FLAGS
+
+// Convert the value under x into a string
+#define XTOSTRING(x) #x
+// Evaluate the flag value; without this step, it would stringify the flag name
+// itself instead of the flag value
+#define TOSTRING(x) XTOSTRING(x)
+
+  static constexpr bool getFlag(const std::string_view &name) {
+    std::string nameStr = name.data();
+    std::string featureFlags = TOSTRING(REANIMATED_FEATURE_FLAGS);
+    if (featureFlags.find("[" + nameStr + ":") == std::string::npos) {
+      throw std::logic_error("Unable to recognize flag: " + nameStr);
+    }
+    return featureFlags.find("[" + nameStr + ":true]") != std::string::npos;
+  }
+
+#else
+
+  static constexpr bool getFlag(const char *) {
```

Comment:
If you use c++ at least c++20, force this to be compile time evaluated. If you do not support c++20, ignore this comment.
```suggestion
  static consteval bool getFlag(const char *) {
```